Source/Core/UpdaterCommon/UpdaterCommon.cpp

```diff
@@ -747,10 +720,7 @@ bool RunUpdater(std::vector<std::string> args)
   TodoList todo = ComputeActionsToDo(this_manifest, next_manifest);
   todo.Log();
 
-  std::optional<std::string> maybe_temp_dir = FindOrCreateTempDir(opts.install_base_path);
-  if (!maybe_temp_dir)
-    return false;
-  std::string temp_dir = std::move(*maybe_temp_dir);
+  std::string temp_dir = File::CreateTempDir();
```

Comment:
I'm sorry, there is a behavioural change I missed during my last review... CreateTempDir can fail, in which case it will return an empty string. The return value should be checked to avoid littering the current directory with update files when CreateTempDir fails.